src/js_realm.hpp

```diff
@@ -785,17 +787,100 @@ bool RealmClass<T>::get_realm_config(ContextType ctx, size_t argc, const ValueTy
+ * allows users to specify an initial set of flexible sync subscriptions to be
+ * bootstrapped when opening a Realm.
+ *
+ * If an `initialSubscriptions` object is provided in the config, it must be an
+ * object with an `update` property, which is a function (called with the Realm
+ * as an argument) which should update the Realm's subscriptions. If called from
+ * `Realm.open`, then the JS (in the `open` method in `lib/extensions.js`) will
+ * return a promise which resolves when the subscription update has been
+ * synchronised.
+ *
+ * If the object contains a `rerunOnStartup` property, this must be a boolean,
+ * which specifies that we should re-run the `update` function every time the
+ * Realm is opened rather than just the first time. This allows users to
+ * workaround the lack of "dynamic" date queries for synced data (e.g. "last 30
+ * days"), by creating the dynamic values in their lanaguge and passing them
+ * into the subscription's, which can then be updated (by using a named query)
+ * every time their app starts.
+ *
+ * This method is separate from `get_realm_config` because this functionality is
+ * not yet implemented in core, and the work needs to performed after the Realm
+ * has been opened in its own write transaction so `initialization_function` is
+ * not suitable.
+ *
+ * TODO remove this once functionality is implemented in core.
+ *
+ * @tparam T The JavaScript engine
+ * @param ctx The JavaScript context
+ * @param argc Number of arguments
+ * @param arguments Array of arguments (contains the config, if any)
+ * @param realm_object The Realm on which to subscribe, wrapped as an ObjectType
+ * @param realm_exists Boolean specifying whether the Realm already existed at
+ * the time of opening it or not
+ */
+template <typename T>
+void RealmClass<T>::handle_initial_subscriptions(ContextType ctx, size_t argc, const ValueType arguments[],
+                                                 ObjectType realm_object, bool realm_exists)
+{
+    if (argc == 0) {
+        return;
+    }
+
+    ValueType config_value = arguments[0];
+    if (!Value::is_object(ctx, config_value)) {
+        return;
+    }
+    ObjectType config_object = Value::validated_to_object(ctx, config_value);
+
+    ValueType sync_value = Object::get_property(ctx, config_object, "sync");
+    if (Value::is_undefined(ctx, sync_value)) {
+        return;
+    }
+    ObjectType sync_object = Value::validated_to_object(ctx, sync_value);
+
+    ValueType initial_subscriptions_value = Object::get_property(ctx, sync_object, "initialSubscriptions");
+    if (Value::is_undefined(ctx, initial_subscriptions_value)) {
+        return;
+    }
+    ObjectType initial_subscriptions_object = Value::validated_to_object(ctx, initial_subscriptions_value);
+
+    ValueType update_value = Object::get_property(ctx, initial_subscriptions_object, "update");
+    FunctionType update_function = Value::validated_to_function(ctx, update_value, "update");
```

Comment:
`NIT`: Checking if `update_value` is undefined might be a good idea

Reply:
I believe this is already handled by `validated_to_function`, I've added a test for this case which passes (not yet pushed)